src/preprocessing/preprocessing-modules/HTMLTag.ts

```diff
@@ -0,0 +1,37 @@
+import { Config } from '../../config';
+import { parseHTML, serializeHTML } from '../util';
+
+const HTML_ATTRIBUTES = [
+  'allow-xhr-interception',
+  'allow-viewer-render-template',
+  'report-errors-to-viewer',
+];
+
+/**
+ * Adds attributes to the <html> tag which affect which features are allowed by
+ * the AMP viewer.
+ *
+ * @param {string} amp AMP code to modify <html> tag of
+ * @param {!Config} config Global config
+ * @return {string} Modified AMP code
+ */
+function process(amp: string, config: Config): string {
+  const doc = parseHTML(amp);
+
+  const tags = doc.getElementsByTagName('html');
+  if (tags.length !== 1) {
+    throw new Error('Failed to find <html> tag inside AMP document');
```

Comment:
The validator should catch this error. We should validate the doc prior to applying this html tag module. But herein lies the issue where if the opt out of that process then this wouldn't be caught, so I guess it's fine to keep.

Also `doc.documentElement` would also be the html root tag, which is more concise than `doc.getElementsByTagName`

Reply:
Yeah, this is more of a "should never happen" kind of exception, I don't know many cases where there can be more or less than one `<html>` but I added it just in case as a precondition.

I decided to avoid `doc.documentElement` due to the possibility of DOM clobbering: if the document contains e.g. `<img name=documentElement>` then `doc.documentElement`. Technically, you could replace `getElementsByTagName` too this way, but it'd at least throw an exception then.